extensions/amp-auto-ads/0.1/firstimpression.io-network-config.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+import {buildUrl} from '../../../ads/google/a4a/shared/url-builder';
+import {dict} from '../../../src/utils/object';
+
+/**
+ * @implements {./ad-network-config.AdNetworkConfigDef}
+ */
+export class FirstImpressionIoConfig {
+  /**
+   * @param {!Element} autoAmpAdsElement
+   */
+  constructor(autoAmpAdsElement) {
+    this.autoAmpAdsElement_ = autoAmpAdsElement;
+  }
+
+  /**
+   * @param {!Window} unused
+   * @override
+   */
+  isEnabled(unused) {
+    return true;
+  }
+
+  /**
+   * True if responsive is enabled for auto-ads
+   */
+  isResponsiveEnabled() {
+    return false;
+  }
+
+  /** @override */
+  getConfigUrl() {
+    let queryParams = {};
+    let previewId = 0;
+
+    const previewFlowRegex = /amp\/fi\/(\d+)\//;
```

Comment:
I don't feel strongly, but would a query param be easier to parse?

Reply:
This is specific to how our other system works. This regex will work in the flow when an AMP page is opened from our AMP proxy. The AMP proxy works with URL structure like: /amp/fi/1234/http://example.com/index.html. To avoid any clashes with target page's URL parameters the URL prefix is used.

Reply:
ok this is fine.